### PR TITLE
Resolve #7 - New Async Void analyzer

### LINKED ISSUE #7: Add analyzer for `async void` method signatures.
These instances are almost always going to be unintentional. They cause the calling thread to fire and forget which can lead to some subtle and difficult to track down bugs, one of which I just recently dealt with.

Solution: flag them as warnings or even errors and recommend changing to `async Task`

### PR DESCRIPTION
**Pull request by brogowski**
Hi,

Here is new analyzer to look after async void method declarations.

### CODE REVIEW

**Comment by Keboo** [comment]
Fixes #7

File: IntelliTectAnalyzer/IntelliTectAnalyzer/Analyzers/AsyncVoid.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace IntelliTectAnalyzer.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class AsyncVoid : DiagnosticAnalyzer
+    {
+        public const string DiagnosticId = "INTL0004";
```
Comment (Keboo) [inline IntelliTectAnalyzer/IntelliTectAnalyzer/Analyzers/AsyncVoid.cs]: Can we move this to a new ID `INTL0201`

Comment (brogowski) [reply IntelliTectAnalyzer/IntelliTectAnalyzer/Analyzers/AsyncVoid.cs]: Sure, no problem

File: IntelliTectAnalyzer/IntelliTectAnalyzer.Test/AsyncVoidTests.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    namespace ConsoleApplication1
+    {
+        class TypeName
+        {   
+            public async void Sample() { }
+        }
+    }";
+            var expected = new DiagnosticResult
+            {
+                Id = "INTL0004",
```
Comment (Keboo) [inline IntelliTectAnalyzer/IntelliTectAnalyzer.Test/AsyncVoidTests.cs]: Looks like the unit test needs to be updated to match.

Comment (brogowski) [reply IntelliTectAnalyzer/IntelliTectAnalyzer.Test/AsyncVoidTests.cs]: Oh, darn - forgot to run them, my bad...

**Comment by Keboo** [comment]
Awesome work. Thank you!

**Pull request merged**